website/src/edit-word-panel.tsx

```diff
@@ -0,0 +1,127 @@
+import React from "react"
+import { GoPencil } from "react-icons/go"
+import { IoCheckmark } from "react-icons/io5"
+import { Button } from "reakit/Button"
+import {
+  unstable_Form as Form,
+  unstable_FormInput as FormInput,
+  unstable_FormLabel as FormLabel,
+} from "reakit/Form"
+import { unstable_FormSubmitButton as FormSubmitButton } from "reakit/Form"
+import {
+  buttonStyle,
+  cleanButtonStyle,
+  formInputLabelStyle,
+  formStyle,
+  inputStyle,
+  textareaStyle,
+} from "./edit-word-panel.css"
+import { useForm } from "./form-context"
+
+/** Button that allows user to enter edit mode in the word panel, and edit fields of a word. */
+export const EditButton = () => {
+  const { form, isEditing, setIsEditing } = useForm()
+
+  return (
+    <Form {...form} className={formStyle}>
+      {isEditing ? (
+        // Displays a "Cancel" button and "Save" button in editing mode.
+        <>
+          <Button
+            className={cleanButtonStyle}
+            aria-label="Cancel editing the selected word and its features"
+            onClick={() => {
+              setIsEditing(false)
+            }}
+          >
+            Cancel
+          </Button>
+
+          <FormSubmitButton
+            {...form}
+            className={buttonStyle}
+            aria-label="Save the selected word and its features"
+          >
+            <IoCheckmark />
+            Save
+          </FormSubmitButton>
+        </>
+      ) : (
+        <Button
+          className={buttonStyle}
+          onClick={() => {
+            setIsEditing(true)
+          }}
+          aria-label="Edit the selected word and its features"
+        >
+          <GoPencil />
+          Edit
+        </Button>
+      )}
+    </Form>
+  )
+}
+
+/** Displays a FormInput with its corresponding feature data from the Reakit form. */
+const EditWordPanel = (props: { features: string[] }) => {
+  const { form } = useForm()
+
+  if (!form || !form.values.word) {
+    return <></>
+  }
+
+  // To retrieve data about a word's field, it's grabbed from the form using the key. All keys begin with "word".
+  const name = ["word"].concat(props.features)
+
+  let formInput = <FormInput {...form} className={inputStyle} name={name} />
+
+  let label = undefined
+
+  switch (name[1]) {
+    case "commentary":
+      // If the current feature is commentary, then display a bigger text area by default.
+      formInput = (
+        <FormInput
+          {...form}
+          as="textarea"
+          rows={3}
+          className={textareaStyle}
+          name={name}
+        />
+      )
+      break
+    case "source":
+      label = "Syllabary Characters"
+      break
+    case "simplePhonetics":
+      label = "Simple Phonetics"
+      break
+    case "romanizedSource":
+      label = "Romanized Source"
+      break
+    case "englishGloss":
+      label = "English Translation"
+      break
+    default:
```

Comment:
You've got high cohesion between your behavior and your copy/data which means a higher maintenance burden. I shouldn't need to edit the general field editor component to change the label of a specific field. I would make `label` a prop for sure.

Reply:
Got it thanks, went ahead and added that as a prop!